Sources/ArcGISToolkit/FloorFilter/SiteFacilityPromptViewController.swift

```diff
@@ -0,0 +1,264 @@
+            backBtn?.isHidden = true
+        }
+    }
+}
+
+/// Extension for Search Bar functions
+extension SiteFacilityPromptViewController: UISearchBarDelegate {
+    private func initializeSiteFacilitySearchBar() {
+        siteFacilitySearchBar.delegate = self
+        resetSearchFilteredResults()
+        siteFacilitySearchBar.tintColor = .customBlue
+    }
+    
+    public func searchBarTextDidBeginEditing(_ searchBar: UISearchBar) {
+        isSearchActive = true
+        self.siteFacilitySearchBar.showsCancelButton = true
+    }
+    
+    public func searchBarSearchButtonClicked(_ searchBar: UISearchBar) {
+        isSearchActive = false
+        self.siteFacilitySearchBar.resignFirstResponder()
+        self.siteFacilitySearchBar.endEditing(true)
+    }
+    
+    public func searchBarCancelButtonClicked(_ searchBar: UISearchBar) {
+        isSearchActive = false
+        self.siteFacilitySearchBar.text = ""
+        self.siteFacilitySearchBar.resignFirstResponder()
+        self.siteFacilitySearchBar.endEditing(true)
+    }
+    
+    public func searchBarTextDidEndEditing(_ searchBar: UISearchBar) {
+        isSearchActive = false
+        self.siteFacilitySearchBar.resignFirstResponder()
+        self.siteFacilitySearchBar.endEditing(true)
+    }
+    
+    private func resetSearchFilteredResults() {
+        filteredSearchSites = viewModel.sites
+        filteredSearchFacilities = viewModel.facilities
+    }
+    
+    private func dismissSearchBar() {
+        self.isSearchActive = false
+        self.siteFacilitySearchBar.text = ""
+        self.siteFacilitySearchBar.resignFirstResponder()
+        self.siteFacilitySearchBar.endEditing(true)
+        self.siteFacilitySearchBar.showsCancelButton = false
+    }
+}
+
+/// Extension for the Sites and Facilities Table View
+extension SiteFacilityPromptViewController: UITableViewDataSource, UITableViewDelegate {
+    private func initializeSiteFacilityTableView() {
+        siteFacilityTableView.delegate = self
+        siteFacilityTableView.dataSource = self
+        siteFacilityTableView.separatorStyle = .none
+        siteFacilityTableView.register(UINib(nibName: "FloorFilterSiteFacilityCell", bundle: .module), forCellReuseIdentifier: "FloorFilterSiteFacilityCell")
+    }
+    
+    public func tableView(_ tableView: UITableView, numberOfRowsInSection section: Int) -> Int {
+        updatePromptTitle()
+        return isShowingFacilities ? filteredFacilities().count : filteredSites().count
+    }
+    
+    public func tableView(_ tableView: UITableView, cellForRowAt indexPath: IndexPath) -> UITableViewCell {
+        let cell = siteFacilityTableView.dequeueReusableCell(withIdentifier: "FloorFilterSiteFacilityCell", for: indexPath) as! SiteFacilityTableViewCell
+        let sites = filteredSites()
+        let facilities = filteredFacilities()
+        
+        // If there are no sites in the map, then directly show the facilities list
+        if (sites.isEmpty) {
```

Comment:
Do you also want to directly show the facilities if there's only one site?

Reply:
@mhdostal in an email thread with @tomhillgeo and the rest of the team, we decided to still show the sites even if there is one site. I'll add a property to hide site button in the viewcontroller so the developer can hide it.

Reply:
@mhdostal, for a future enhancement, we can directly show the facilities if there is only site and show the site name on the prompt so the user knows what site is selected.